src/react-count-to.js

```diff
@@ -48,23 +48,30 @@ const CountTo = React.createClass({
   start() {
     this.clear();
     this.setState(this.getInitialState(), () => {
-      const delay = this.props.delay;
+      const { delay, speed, to } = this.props;
+      const { counter } = this.state;
       this.loopsCounter = 0;
-      this.loops = Math.ceil(this.props.speed / delay);
-      this.increment = (this.props.to - this.state.counter) / this.loops;
+      this.loops = Math.ceil(speed / delay);
+      this.increment = (to - counter) / this.loops;
       this.interval = setInterval(this.next, delay);
     });
   },
 
   next() {
     if (this.loopsCounter < this.loops) {
+      let { counter } = this.state;
+      counter = counter + this.increment;
```

Comment:
Yeah, in this particular case is not a problem because you are mutating a primitive but it's always a good practice to avoid doing that.
If you don't mind I would ask you to change it, thanks!